src/circuits/atomicSigV2.ts

```diff
@@ -0,0 +1,153 @@
+    fieldIdx++;
+
+    //  - userID
+    this.userId = Id.fromBigInt(BigInt(pubSignals[fieldIdx]));
+    fieldIdx++;
+
+    // - issuerAuthState
+    this.issuerAuthState = newHashFromString(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - requestID
+    this.challenge = BigInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - issuerID
+    this.issuerID = Id.fromBigInt(BigInt(pubSignals[fieldIdx]));
+    fieldIdx++;
+
+    this.isRevocationChecked = parseInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - issuerClaimNonRevState
+    this.issuerClaimNonRevState = newHashFromString(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    //  - timestamp
+    this.timestamp = parseInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    //  - claimSchema
+    this.claimSchema = SchemaHash.newSchemaHashFromInt(
+      BigInt(pubSignals[fieldIdx]),
+    );
+    fieldIdx++;
+
+    // - ClaimPathNotExists
+    this.claimPathNotExists = parseInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - ClaimPathKey
+    this.claimPathKey = BigInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - slotIndex
+    this.slotIndex = parseInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    // - operator
+    this.operator = parseInt(pubSignals[fieldIdx]);
+    fieldIdx++;
+
+    //  - values
+    for (let index = 0; index < valuesSize; index++) {
+      this.value.push(BigInt(pubSignals[fieldIdx]));
+      fieldIdx++;
+    }
+  }
+
+  async verifyQuery(query: Query, schemaLoader: ISchemaLoader): Promise<void> {
+    const outs: ClaimOutputs = {
+      issuerId: this.issuerID,
+      schemaHash: this.claimSchema,
+      slotIndex: this.slotIndex,
+      operator: this.operator,
+      value: this.value,
+      timestamp: this.timestamp,
+      merklized: this.merklized,
+      claimPathKey: this.claimPathKey,
+      claimPathNotExists: this.claimPathNotExists,
+      valueArraySize: valuesSize,
+      isRevocationChecked: this.isRevocationChecked,
+    };
+    return await checkQueryRequest(query, outs, schemaLoader);
+  }
+  async verifyStates(resolvers: Resolvers, opts?: VerifyOpts): Promise<void> {
+    const resolver = getResolverByID(resolvers, this.issuerID);
+    if (resolver === undefined) {
+      throw new Error(`resolver not found for issuerID ${this.issuerID}`);
+    }
+
+    await checkUserState(resolver, this.issuerID, this.issuerAuthState);
+
+    if (this.isRevocationChecked === 0) {
+      return;
+    }
+
+    const issuerNonRevStateResolved = await checkIssuerNonRevState(
+      resolver,
+      this.issuerID,
+      this.issuerClaimNonRevState,
+    );
+
+    let acceptedStateTransitionDelay = defaultProofVerifyOpts;
+    if (!!opts && !!opts.AcceptedStateTransitionDelay) {
+      acceptedStateTransitionDelay = Number(opts.AcceptedStateTransitionDelay);
+    }
+
+    if (!issuerNonRevStateResolved.latest) {
+      const timeDiff =
+        Date.now() - Number(issuerNonRevStateResolved.transitionTimestamp);
```

Comment:
transitionTimestamp is unix format?

Reply:
Yes, is UNIX format. https://ethereum.stackexchange.com/questions/18192/how-do-you-work-with-date-and-time-on-ethereum-platform

Reply:
in that case, you probably need to convert the UNIX timestamp to date. You can use https://github.com/iden3/js-iden3-core/blob/main/src/utils.ts#L63

Reply:
Done